### PR TITLE
[CBRD-20577] Use of alias with function inside ‘Order By’ Clause

### PR DESCRIPTION
**Pull request by byungwook-kim**
http://jira.cubrid.org/browse/CBRD-20577

**The order by clause should allow the alias name specified at select list as ORACLE.**

The function **pt_resolve_names** calls **parser_walk_tree with pt_mark_group_having_pt_name**.
And  the function **pt_mark_group_having_pt_name** calls again _parser_walk_tree_ with _node->info.query.q.select.group_by_ and _node->info.query.q.select.having_ for resolving name resolutions.

We need to add parser_walk_tree with _node->info.query.order_by_.

As the same reason, we need to add the loop by **pt_resolve_group_having_alias_internal** with _node->info.query.order_by_ in the function **pt_resolve_group_having_alias**.

### CODE REVIEW

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/parser/name_resolution.c (9 incorrect indentation(s) found)
```
  7790   if (!PT_SELECT_INFO_IS_FLAGED (node, PT_SELECT_INFO_COLS_SCHEMA | PT_SELECT_FULL_INFO_COLS_SCHEMA))
  7794 	{
  7795 	  node->info.query.order_by =
  7797 	}
  7999   if (!PT_SELECT_INFO_IS_FLAGED (node, PT_SELECT_INFO_COLS_SCHEMA | PT_SELECT_FULL_INFO_COLS_SCHEMA))
  8003 	{
  8004 	  pt_resolve_group_having_alias_internal (parser, &pt_cur, node->info.query.q.select.list);
  8005 	  pt_cur = pt_cur->next;
  8006 	}
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/parser/name_resolution.c (8 incorrect indentation(s) found)
```
  7793 	{
  7794 	  node->info.query.order_by =
  7795 	    parser_walk_tree (parser, node->info.query.order_by, pt_mark_pt_name, NULL, NULL, NULL);
  7796 	}
  8002 	{
  8003 	  pt_resolve_group_having_alias_internal (parser, &pt_cur, node->info.query.q.select.list);
  8004 	  pt_cur = pt_cur->next;
  8005 	}
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

File: src/parser/name_resolution.c [hunk 1]
```diff
@@ -7787,6 +7787,15 @@ pt_mark_group_having_pt_name (PARSER_CONTEXT * parser, PT_NODE * node, void *chk
 	parser_walk_tree (parser, node->info.query.q.select.having, pt_mark_pt_name, NULL, NULL, NULL);
     }
 
+  if (!PT_SELECT_INFO_IS_FLAGED (node, PT_SELECT_INFO_COLS_SCHEMA | PT_SELECT_FULL_INFO_COLS_SCHEMA))
```
Comment (shparkcubrid) [inline src/parser/name_resolution.c:7790]: what about including 'PT_SELECT_INFO_IDX_SCHEMA'?

Comment (byungwook-kim) [reply src/parser/name_resolution.c:7790]: 'show columns ..." statement uses alias but "show index ... " doesn't use alias.

**Pull request merged**